src/next-reminders/lib/db.test.ts

```diff
@@ -0,0 +1,171 @@
+import { OneOffSignup } from '../../create-reminder-signup/lambda/models';
+import { createDatabaseConnectionPool, DBConfig } from '../../lib/db';
+import { getNextReminders } from './db';
+
+const config: DBConfig = {
+	url: process.env.TEST_DB_URL ?? '',
+	username: process.env.TEST_DB_USER ?? '',
+	password: process.env.TEST_DB_PASSWORD ?? '',
+};
+
+const pool = createDatabaseConnectionPool(config);
+
+const createOneOffReminder = ({
+	identityId = '0',
+	country = 'GB',
+	reminderPeriod = '2021-01-01',
+	reminderCreatedAt = '2021-01-01',
+	reminderPlatform = 'WEB',
+	reminderComponent = 'EPIC',
+	reminderStage = 'PRE',
+}): OneOffSignup => ({
+	identity_id: identityId,
+	country: country,
+	reminder_period: reminderPeriod,
+	reminder_created_at: reminderCreatedAt,
+	reminder_platform: reminderPlatform,
+	reminder_component: reminderComponent,
+	reminder_stage: reminderStage,
+});
+
+interface SaveOneOffReminderOptions {
+	cancelledAt?: string;
+}
+
+const saveOneOffReminder = (
+	reminder: OneOffSignup,
+	options?: SaveOneOffReminderOptions,
+) => {
+	return pool.query({
+		text: `
+			INSERT INTO one_off_reminder_signups(
+	            identity_id,
+				country,
+	            reminder_period,
+	            reminder_created_at,
+				reminder_cancelled_at,
+	            reminder_platform,
+	            reminder_component,
+	            reminder_stage,
+				reminder_option
+	        ) VALUES (
+	            $1, $2, $3, $4, $5, $6, $7, $8, $9
+	        )
+			`,
+		values: [
+			reminder.identity_id,
+			reminder.country,
+			reminder.reminder_period,
+			reminder.reminder_created_at,
+			options?.cancelledAt,
+			reminder.reminder_platform,
+			reminder.reminder_component,
+			reminder.reminder_stage,
+			reminder.reminder_option,
+		],
+	});
+};
+
+beforeAll(() => {
+	const initDatabase = async (): Promise<void> => {
+		await pool.query(`DROP TABLE IF EXISTS one_off_reminder_signups`);
+		await pool.query(`
+			CREATE TABLE one_off_reminder_signups(
+				identity_id TEXT NOT NULL,
+				country TEXT,
+				reminder_created_at TIMESTAMP NOT NULL,
+				reminder_cancelled_at TIMESTAMP,
+				reminder_platform TEXT NOT NULL,
+				reminder_component TEXT NOT NULL,
+				reminder_stage TEXT NOT NULL,
+				reminder_period DATE NOT NULL,
+				reminder_option TEXT,
+				PRIMARY KEY (identity_id, reminder_period)
+			);
+		`);
+
+		await pool.query(`DROP TABLE IF EXISTS recurring_reminder_signups`);
+		await pool.query(`
+			CREATE TABLE recurring_reminder_signups(
+				identity_id TEXT NOT NULL PRIMARY KEY,
+				country TEXT,
+				reminder_created_at TIMESTAMP NOT NULL,
+				reminder_cancelled_at TIMESTAMP,
+				reminder_platform TEXT NOT NULL,
+				reminder_component TEXT NOT NULL,
+				reminder_stage TEXT NOT NULL,
+				reminder_frequency_months INT NOT NULL,
+				reminder_option TEXT
+			);
+		`);
```

Comment:
We could potentially have this be part of the test set up process (i.e. a dev has to go and create the database + tables) which would save us from having all of this here. However, if we think this is better, I will at least factor this out into a test helpers file to remove some of the noise

Reply:
Perhaps this is the time to move the table creation sql into this repo?

Reply:
Yeah if we did could have all this sql in that file, and then we would just load the file (`fs.readFileSync(/*...*/)`) and execute that string. I can have a go at that refactor? It probably makes sense

Reply:
This has now become:
```js
	const initDatabase = async (): Promise<void> => {
		const query = readFileSync(
			'./sql/create-signups-tables.sql',
		).toString();

		await pool.query(query);
	};
```

which is hard to say no to!